### ISSUE TITLE
 angular/di-unused: does not work with arrow function declaration

### ISSUE THREAD

**Issue by ojab** [Gillespie59/eslint-plugin-angular#443]
Consider the code:
```javascript
angular.module("").factory("", ["$http", "$q", function($http, $q) {
    return $q.resolve();
}])
```
the correct warning `1:48  warning  Unused injected value $http  angular/di-unused` is reported

But if function declaration will be changed to arrow function, i. e.:
```javascript
angular.module("").factory("", ["$http", "$q", ($http, $q) => {
    return $q.resolve();
}])
```
no warning is emitted.

eslint-plugin-angular@1.6.1 and eslint@3.15.0 (with `"es6": true`) is used.

**Comment by 0xR** [comment]
I am running into the same issue, would be great if it could be fixed. I'll have a look if can fix it < 20 min.

**Comment by 0xR** [comment]
In 20 min I managed to make failing unit tests and try some changed that didn't work. My progress is here: https://github.com/0xR/eslint-plugin-angular/commit/aa5d64046abce01cf3509888c0b8028c9f2079dd
